src/infra/account/secureAccessKey/secureAccessKeyCmdRepo.go

```diff
@@ -0,0 +1,202 @@
+	accountUsername valueObject.Username,
+) error {
+	accountUsernameStr := accountUsername.String()
+
+	secureAccessKeysDirPath := "/home/" + accountUsernameStr + "/.ssh"
+	if !infraHelper.FileExists(secureAccessKeysDirPath) {
+		err := infraHelper.MakeDir(secureAccessKeysDirPath)
+		if err != nil {
+			return errors.New("CreateSecureAccessKeysDirectoryError: " + err.Error())
+		}
+	}
+
+	secureAccessKeysFilePath := secureAccessKeysDirPath + "/authorized_keys"
+	if infraHelper.FileExists(secureAccessKeysFilePath) {
+		return nil
+	}
+
+	_, err := os.Create(secureAccessKeysFilePath)
+	if err != nil {
+		return errors.New("CreateSecureAccessKeysFileError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmd(
+		"chown", "-R", accountUsernameStr, secureAccessKeysFilePath,
+	)
+	if err != nil {
+		return errors.New("ChownSecureAccessKeysFileError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *SecureAccessKeyCmdRepo) allowAccountSecureRemoteConnection(
+	accountId valueObject.AccountId,
+) error {
+	accountUsername, err := infraHelper.RunCmdWithSubShell(
+		"awk -F: '$3 == " + accountId.String() +
+			" && $7 != \"/bin/bash\" {print $1}' /etc/passwd",
+	)
+	if err != nil {
+		return errors.New("ReadUnixUsernameFromFileError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmdWithSubShell(
+		"chsh -s /bin/bash " + accountUsername,
+	)
+	if err != nil {
+		return errors.New("ChangeDefaultBashError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *SecureAccessKeyCmdRepo) recreateSecureAccessKeysFile(
+	accountId valueObject.AccountId,
+	accountUsername valueObject.Username,
+) error {
+	keysFilePath := "/home/" + accountUsername.String() + "/.ssh/authorized_keys"
+	if !infraHelper.FileExists(keysFilePath) {
+		err := repo.createSecureAccessKeysFileIfNotExists(accountUsername)
+		if err != nil {
+			return err
+		}
+	}
+
+	readRequestDto := dto.ReadSecureAccessKeysRequest{
+		Pagination: dto.Pagination{
+			ItemsPerPage: 1000,
+		},
+		AccountId: accountId,
+	}
+	keys, err := repo.secureAccessKeyQueryRepo.Read(readRequestDto)
+	if err != nil {
+		return err
+	}
+
+	keysFileContent := ""
+	for _, key := range keys.SecureAccessKeys {
+		keysFileContent += key.Content.String() + " " + key.Name.String() + "\n"
+	}
+
+	shouldOverwrite := true
+	err = infraHelper.UpdateFile(keysFilePath, keysFileContent, shouldOverwrite)
+	if err != nil {
+		return errors.New("UpdateSecureAccessKeysFileContentError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *SecureAccessKeyCmdRepo) Create(
+	createDto dto.CreateSecureAccessKey,
+) (keyId valueObject.SecureAccessKeyId, err error) {
+	account, err := repo.accountQueryRepo.ReadById(createDto.AccountId)
+	if err != nil {
+		return keyId, errors.New("AccountNotFound")
+	}
+
+	err = repo.createSecureAccessKeysFileIfNotExists(account.Username)
+	if err != nil {
+		return keyId, err
+	}
+
+	keyContentStr := createDto.Content.String()
+	rawFingerprint, err := infraHelper.RunCmdWithSubShell(
```

Comment:
Why isn't that a getter on the key entity? I think the fingerprint should be a field of the keys.